File: java/src/main/java/com/techelevator/entity/Recipe.java
package com.techelevator.entity;

import javax.persistence.*;

@Entity
@Table(name = "recipe")
public class Recipe {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "recipe_id", nullable = false)
    private Integer id;

    @Column(name = "title", nullable = false, length = 50)
    private String title;

    @Column(name = "details", nullable = false, length = 2000)
    private String details;

    @Column(name = "instructions", nullable = false, length = 2000)
    private String instructions;

    @Column(name = "dish_type", length = 50)
    private String dishType;

    @Column(name = "ispublic", nullable = false)
    private Boolean ispublic = false;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDetails() {
        return details;
    }

    public void setDetails(String details) {
        this.details = details;
    }

    public String getInstructions() {
        return instructions;
    }

    public void setInstructions(String instructions) {
        this.instructions = instructions;
    }

    public String getDishType() {
        return dishType;
    }

    public void setDishType(String dishType) {
        this.dishType = dishType;
    }

    public Boolean getIspublic() {
        return ispublic;
    }

    public void setIspublic(Boolean ispublic) {
        this.ispublic = ispublic;
    }

}

File: frontend/src/Redux/recipe.js
import * as ActionTypes from './actionTypes'

export const Recipe = (state = {
        id: null,
        title: "",
        details: "",
        instructions: "",
        dish_type: "",
        isPublic: false
    }, action) => {
        switch (action.type) {
            case ActionTypes.ADD_RECIPE:
                return {...state,
                    id: action.payload.id,
                    title: action.payload.title,
                    details: action.payload.details,
                    instructions: action.payload.instructions,
                    dishType: action.payload.dishType,
                };

            case ActionTypes.FETCH_RECIPE:
                return action.payload;

            case ActionTypes.EDIT_RECIPE:
                return state.map(recipe => 
                    recipe.id !== action.payload.id ? recipe : {...recipe, ...action.payload});

            case ActionTypes.DELETE_RECIPE:
                return state.filter();

            default:
                return state;
    }

}